content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md

```diff
@@ -9,40 +9,49 @@ weight: 20
+resource to your Kubernetes cluster and when to use a standalone service. It describes the two
+methods for adding custom resources and how to choose between them.
 
 {{% /capture %}}
 
 {{% capture body %}}
 ## Custom resources
 
-A *resource* is an endpoint in the [Kubernetes API](/docs/reference/using-api/api-overview/) that stores a collection of [API objects](/docs/concepts/overview/working-with-objects/kubernetes-objects/) of a certain kind. For example, the built-in *pods* resource contains a collection of Pod objects.
+A *resource* is an endpoint in the [Kubernetes API](/docs/reference/using-api/api-overview/) that stores a collection of
+[API objects](/docs/concepts/overview/working-with-objects/kubernetes-objects/) of a certain kind. For example, the built-in *pods* resource contains a collection of Pod objects.
 
-A *custom resource* is an extension of the Kubernetes API that is not necessarily available on every
-Kubernetes cluster.
-In other words, it represents a customization of a particular Kubernetes installation.
+A *custom resource* is an extension of the Kubernetes API that is not necessarily available in a default
+Kubernetes installation. It represents a customization of a particular Kubernetes installation. However,
+many core Kubernetes functions are now built using custom resources, making Kubernetes more modular.
 
 Custom resources can appear and disappear in a running cluster through dynamic registration,
 and cluster admins can update custom resources independently of the cluster itself.
-Once a custom resource is installed, users can create and access its objects with
-[kubectl](/docs/user-guide/kubectl-overview/), just as they do for built-in resources like *pods*.
+Once a custom resource is installed, users can create and access its objects using the
+[kubectl](/docs/user-guide/kubectl-overview/) command, just as they do for built-in resources like
+*Pods*.
 
-### Custom controllers
+## Custom controllers
 
 On their own, custom resources simply let you store and retrieve structured data.
-It is only when combined with a *controller* that they become a true declarative API.
+When you combine a custom resource with a *custom controller*, custom resources
+provide a true _declarative API_.
+
 A [declarative API](/docs/concepts/overview/working-with-objects/kubernetes-objects/#understanding-kubernetes-objects)
 allows you to _declare_ or specify the desired state of your resource and tries
-to match the actual state to this desired state.
-Here, the controller interprets the structured data as a record of the user's
-desired state, and continually takes action to achieve and maintain this state.
+keep the current state of Kubernetes objects in sync with the desired state.
+The controller interprets the structured data as a record of the user's
+desired state, and continually maintains this state (or generates errors if it
+cannot maintain the state).
```

Comment:
I don't think it is completely accurate to say that it "generate errors" because sometimes if it can't maintain the state, the controller will update the "status" to reflect the current state - but not necessarily fail with an error. I think we can remove the last part in the brackets.